telicent-graph-schema/src/test/java/io/telicent/jena/graphql/fetchers/telicent/graph/TestFilterParsing.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = ".*empty values.*")
+    public void givenFilterWithMissingValues_whenParsing_thenIllegalArgument() {
+        // Given
+        Map<String, Object> arguments = Map.of(TelicentGraphSchema.ARGUMENT_TYPE_FILTER,
+                                               Collections.emptyMap());
+        DataFetchingEnvironment environment = prepareEnvironment(arguments);
+
+        // When and Then
+        this.parseFilter(environment, TelicentGraphSchema.ARGUMENT_TYPE_FILTER);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = ".*non-String values")
+    public void givenFilterWithMalformedValues_whenParsing_thenIllegalArgument() {
+        // Given
+        Map<String, Object> arguments = Map.of(TelicentGraphSchema.ARGUMENT_TYPE_FILTER,
+                                               Map.of(TelicentGraphSchema.ARGUMENT_VALUES, List.of(1, 2, 3)));
+        DataFetchingEnvironment environment = prepareEnvironment(arguments);
+
+        // When and Then
+        this.parseFilter(environment, TelicentGraphSchema.ARGUMENT_TYPE_FILTER);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = ".*empty values.*")
+    public void givenFilterWithNonListValues_whenParsing_thenIllegalArgument() {
+        // Given
+        Map<String, Object> arguments = Map.of(TelicentGraphSchema.ARGUMENT_TYPE_FILTER,
+                                               Map.of(TelicentGraphSchema.ARGUMENT_VALUES, "foo"));
+        DataFetchingEnvironment environment = prepareEnvironment(arguments);
+
+        // When and Then
+        this.parseFilter(environment, TelicentGraphSchema.ARGUMENT_TYPE_FILTER);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = ".*empty values.*")
+    public void givenFilterWithEmptyValues_whenParsing_thenIllegalArgument() {
+        // Given
+        Map<String, Object> arguments = Map.of(TelicentGraphSchema.ARGUMENT_TYPE_FILTER,
+                                               Map.of(TelicentGraphSchema.ARGUMENT_VALUES, Collections.emptyList()));
+        DataFetchingEnvironment environment = prepareEnvironment(arguments);
+
+        // When and Then
+        this.parseFilter(environment, TelicentGraphSchema.ARGUMENT_TYPE_FILTER);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = "Unknown filter argument: foo")
+    public void givenFilterWithBadArgument_whenParsing_thenIllegalArgument() {
+        // Given
+        Map<String, Object> arguments = Map.of("foo",
+                                               Map.of(TelicentGraphSchema.ARGUMENT_VALUES, List.of("test")));
+        DataFetchingEnvironment environment = prepareEnvironment(arguments);
+
+        // When and Then
+        this.parseFilter(environment, "foo");
+    }
```

Comment:
To extend the coverage to be complete. 

```suggestion
    }
    
    @Test
    public void givenTypeFilterWith_filtersDisabled_returnAll() {
        // Given
        // When
        AbstractFilter result = this.createTypeFilter(null, null);
        // Then
        Assert.assertEquals(IncludeAllFilter.INSTANCE, result);
    }

    private static class TestTypePagingFetcher extends TestFilterParsing {
        @Override
        protected boolean enableFilters() {
            return true;
        }
    }
    
    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = "Fetcher implementation MUST override createTypeFilter\\(\\) method to enable type filtering support")
    public void givenTypeFilterWith_filtersEnabled__thenIllegalArgument() {
        // Given
        TestFilterParsing test = new TestTypePagingFetcher();
        // When
        // Then
        test.createTypeFilter(null, null);
    }

```